net/mwan3/files/lib/mwan3/mwan3.sh

```diff
@@ -391,148 +391,136 @@ mwan3_create_iface_iptables()
-			$IPT6 -N mwan3_iface_in_$1
+		if ! $IPT6 -S "mwan3_iface_in_$1" &> /dev/null; then
+			$IPT6 -N "mwan3_iface_in_$1"
 		fi
 
-		$IPT6 -F mwan3_iface_in_$1
-		$IPT6 -A mwan3_iface_in_$1 -i $2 \
+		$IPT6 -F "mwan3_iface_in_$1"
+		$IPT6 -A "mwan3_iface_in_$1" -i "$2" \
 			-m set --match-set mwan3_connected_v6 src \
 			-m mark --mark 0x0/$MMX_MASK \
 			-m comment --comment "default" \
 			-j MARK --set-xmark $MMX_DEFAULT/$MMX_MASK
-		$IPT6 -A mwan3_iface_in_$1 -i $2 -m mark --mark 0x0/$MMX_MASK \
+		$IPT6 -A "mwan3_iface_in_$1" -i "$2" -m mark --mark 0x0/$MMX_MASK \
 			-m comment --comment "$1" \
 			-j MARK --set-xmark $(mwan3_id2mask id MMX_MASK)/$MMX_MASK
 
 		$IPT6 -D mwan3_ifaces_in \
 			-m mark --mark 0x0/$MMX_MASK \
-			-j mwan3_iface_in_$1 &> /dev/null
+			-j "mwan3_iface_in_$1" &> /dev/null
 		$IPT6 -A mwan3_ifaces_in \
 			-m mark --mark 0x0/$MMX_MASK \
-			-j mwan3_iface_in_$1
+			-j "mwan3_iface_in_$1"
 	fi
 }
 
 mwan3_delete_iface_iptables()
 {
-	config_get family $1 family ipv4
+	config_get family "$1" family ipv4
 
-	if [ "$family" == "ipv4" ]; then
+	if [ "$family" = "ipv4" ]; then
 
 		$IPT4 -D mwan3_ifaces_in \
 			-m mark --mark 0x0/$MMX_MASK \
-			-j mwan3_iface_in_$1 &> /dev/null
-		$IPT4 -F mwan3_iface_in_$1 &> /dev/null
-		$IPT4 -X mwan3_iface_in_$1 &> /dev/null
+			-j "mwan3_iface_in_$1" &> /dev/null
+		$IPT4 -F "mwan3_iface_in_$1" &> /dev/null
+		$IPT4 -X "mwan3_iface_in_$1" &> /dev/null
 	fi
 
-	if [ "$family" == "ipv6" ]; then
+	if [ "$family" = "ipv6" ]; then
 
 		$IPT6 -D mwan3_ifaces_in \
 			-m mark --mark 0x0/$MMX_MASK \
-			-j mwan3_iface_in_$1 &> /dev/null
-		$IPT6 -F mwan3_iface_in_$1 &> /dev/null
-		$IPT6 -X mwan3_iface_in_$1 &> /dev/null
+			-j "mwan3_iface_in_$1" &> /dev/null
+		$IPT6 -F "mwan3_iface_in_$1" &> /dev/null
+		$IPT6 -X "mwan3_iface_in_$1" &> /dev/null
 	fi
 }
 
 mwan3_create_iface_route()
 {
-	local id route_args metric
+	local id via metric
 
-	config_get family $1 family ipv4
-	mwan3_get_iface_id id $1
+	config_get family "$1" family ipv4
+	mwan3_get_iface_id id "$1"
 
 	[ -n "$id" ] || return 0
 
-	if [ "$family" == "ipv4" ]; then
-		if ubus call network.interface.${1}_4 status &>/dev/null; then
-			network_get_gateway route_args ${1}_4
-		else
-			network_get_gateway route_args $1
-		fi
-
-		if [ -n "$route_args" -a "$route_args" != "0.0.0.0" ]; then
-			route_args="via $route_args"
+	if [ "$family" = "ipv4" ]; then
+		if ubus call "network.interface.${1}_4" status &>/dev/null; then
+			network_get_gateway via "${1}_4"
 		else
-			route_args=""
+			network_get_gateway via "$1"
 		fi
 
-		network_get_metric metric $1
-		if [ -n "$metric" -a "$metric" != "0" ]; then
-			route_args="$route_args metric $metric"
-		fi
+		network_get_metric metric "$1"
 
-		$IP4 route flush table $id
-		$IP4 route add table $id default $route_args dev $2
+		$IP4 route flush table "$id"
+		$IP4 route add table "$id" default \
+				${via:+via} "$via" \
+				${metric:+metric} "$metric" \
```

Comment:
quotes are not needed here and cause problems if metric or via is not set

Reply:
Thanks for pointing that out, I'm gonna fix that right now.
So we can merge my pullrequest and then you could [rebase](https://github.com/openwrt/packages/pull/12814) yours.